hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OrphanMetaDataManagementTask.java

```diff
@@ -187,4 +262,36 @@ public Pair<String, Boolean> reprocess(OMMetadataManager omMetadataManager) {
     return new ImmutablePair<>(getTaskName(), true);
   }
 
+  private List<OmKeyInfo> getPendingDeletionSubFiles(
+      long volumeId, long bucketId, OmKeyInfo parentInfo)
+      throws IOException {
+    List<OmKeyInfo> files = new ArrayList<>();
+    String seekFileInDB =
+        reconOMMetadataManager.getOzonePathKey(volumeId, bucketId,
+            parentInfo.getObjectID(), "");
+
+    Table fileTable = reconOMMetadataManager.getFileTable();
+    try (TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
+             iterator = fileTable.iterator()) {
+
+      iterator.seek(seekFileInDB);
+
+      while (iterator.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> entry = iterator.next();
+        OmKeyInfo fileInfo = entry.getValue();
+        if (!OMFileRequest.isImmediateChild(fileInfo.getParentObjectID(),
+            parentInfo.getObjectID())) {
+          break;
+        }
+        fileInfo.setFileName(fileInfo.getKeyName());
+        String fullKeyPath = OMFileRequest.getAbsolutePath(
+            parentInfo.getKeyName(), fileInfo.getKeyName());
+        fileInfo.setKeyName(fullKeyPath);
+
+        files.add(fileInfo);
```

Comment:
we can just capture objectId only as required, this to reduce memory required at end of iteration.

Reply:
Done the changes.